shared/src/commonMain/kotlin/org/jetbrains/kotlinconf/screens/SpeakerDetailScreen.kt

```diff
@@ -0,0 +1,106 @@
+package org.jetbrains.kotlinconf.screens
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.widthIn
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinconfapp.shared.generated.resources.Res
+import kotlinconfapp.shared.generated.resources.schedule_in_x_minutes
+import kotlinconfapp.shared.generated.resources.speakers_title
+import org.jetbrains.compose.resources.stringResource
+import org.jetbrains.kotlinconf.ScreenWithTitle
+import org.jetbrains.kotlinconf.SessionId
+import org.jetbrains.kotlinconf.SpeakerId
+import org.jetbrains.kotlinconf.ui.components.SpeakerAvatar
+import org.jetbrains.kotlinconf.ui.components.StyledText
+import org.jetbrains.kotlinconf.ui.components.TalkCard
+import org.jetbrains.kotlinconf.ui.components.TalkStatus
+import org.jetbrains.kotlinconf.ui.theme.KotlinConfTheme
+import org.koin.compose.viewmodel.koinViewModel
+import org.koin.core.parameter.parametersOf
+
+@Composable
+fun SpeakerDetailScreen(
+    speakerId: SpeakerId,
+    onBack: () -> Unit,
+    onSession: (SessionId) -> Unit,
+    viewModel: SpeakerDetailViewModel = koinViewModel { parametersOf(speakerId) }
+) {
+    val speaker = viewModel.speaker.collectAsStateWithLifecycle().value
+    val sessions = viewModel.sessions.collectAsStateWithLifecycle().value
+
+    ScreenWithTitle(
+        title = stringResource(Res.string.speakers_title),
+        onBack = onBack,
+    ) {
+        if (speaker != null) {
```

Comment:
Please consider early return

Reply:
We need some 404 placeholder here when speaker is not found for any reason

Reply:
We'll need empty/error screens across the app!

Reply:
https://github.com/JetBrains/kotlinconf-app/issues/281